src/neo/Network/P2P/Payloads/Header.cs

```diff
@@ -1,16 +1,82 @@
+using Neo.IO;
+using Neo.IO.Json;
+using Neo.Ledger;
+using Neo.Persistence;
+using Neo.SmartContract;
+using Neo.SmartContract.Native;
+using Neo.Wallets;
 using System;
 using System.IO;
 
 namespace Neo.Network.P2P.Payloads
 {
-    public class Header : BlockBase, IEquatable<Header>
+    public sealed class Header : IEquatable<Header>, IVerifiable
     {
-        public override int Size => base.Size + 1;
+        public uint Version;
+        public UInt256 PrevHash;
+        public UInt256 MerkleRoot;
+        public ulong Timestamp;
+        public uint Index;
+        public byte PrimaryIndex;
+        public UInt160 NextConsensus;
+        public Witness Witness;
 
-        public override void Deserialize(BinaryReader reader)
+        private UInt256 _hash = null;
+        public UInt256 Hash
         {
-            base.Deserialize(reader);
-            if (reader.ReadByte() != 0) throw new FormatException();
+            get
+            {
+                if (_hash == null)
+                {
+                    _hash = this.CalculateHash();
+                }
+                return _hash;
+            }
+        }
+
+        public int Size =>
+            sizeof(uint) +      // Version
+            UInt256.Length +    // PrevHash
+            UInt256.Length +    // MerkleRoot
+            sizeof(ulong) +     // Timestamp
+            sizeof(uint) +      // Index
+            sizeof(byte) +      // PrimaryIndex
+            UInt160.Length +    // NextConsensus
+            1 + Witness.Size;   // Witness   
```

Comment:
We can even keep the same header size by removing this `1`.

Reply:
We have already discussed this before. 😆